Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -376,23 +376,33 @@ public class PBXProjGenerator {
 
     func sortGroups(group: PBXGroup) {
         // sort children
-        let children = group.children
-            .sorted { child1, child2 in
-                let sortOrder1 = child1.getSortOrder(groupSortPosition: project.options.groupSortPosition)
-                let sortOrder2 = child2.getSortOrder(groupSortPosition: project.options.groupSortPosition)
-
-                if sortOrder1 != sortOrder2 {
-                    return sortOrder1 < sortOrder2
-                } else {
-                    if child1.nameOrPath != child2.nameOrPath {
-                        return child1.nameOrPath.localizedStandardCompare(child2.nameOrPath) == .orderedAscending
+        if let groupOrder = project.options.groupsOrder.first(where: { group.nameOrPath == $0.pattern || group.nameOrPath.isMatch(to: $0.pattern) }) {
+            let files = group.children.filter { $0 is PBXFileReference }
+            let groups = groupOrder.order
```

Comment:
If someone adds a directory or target, it won't be included as a group unless specifically specified in the groupsOrder. Is that intended?

Reply:
No, it was not supposed, I'll think how to fix it.

Reply:
I've updated PR and it works this way.
1. Everything sort as was previously, i.e. using `sortGroups` functions.
2. Setup groups order for provided groups
3. Not provided groups put at the end of the main group.

**Example**

I have a project with a structure:

```
Test.xcodeproj
  Sources
  Resources
  Tests
  Support files
  Configurations
  Products
  Frameworks
```

Xcodegen spec:

```
options:
  groupsOrder: 
    - order: [Sources, Resources, Tests]
```

The result will be looks like:

```
Test.xcodeproj
  Sources
  Resources
  Tests
  Configurations
  Support files
  Products
  Frameworks
```

**Explanation**

First, we sort groups alphabetically, i.e. `Configurations, Resources, Support files, Sources, Tests` after this we setup group order with provided group names `[Sources, Resources, Tests]` and get result `Sources, Resources, Tests` and remaining groups `Configurations, Support files` put at the end.